### ISSUE TITLE
Not reading `pyproject.toml` on Windows?

### ISSUE THREAD

**Issue by ajfriend** [joerick/cibuildwheel#350]
That's for the awesome work on this project! I've been using it to build wheels for [our project here](https://github.com/uber/h3-py).

I'm able to build wheels across Windows/Mac/Linux. However, on Windows I need to specify `CIBW_BEFORE_BUILD: pip install scikit-build cython cmake` to get things to work. That doesn't seem to be necessary on Mac/Linux, as those build requirements are read from the `pyproject.toml` file. 

I'm guessing something is happening in the Windows flow that misses reading `pyproject.toml`?

For reference, my `cibuildwheel` commands for Windows on Github Actions look like:

```yaml
- name: Build wheels
  env:
    CIBW_SKIP: pp* cp27* *-win32
    CIBW_BEFORE_BUILD: pip install scikit-build cython cmake
    CIBW_TEST_REQUIRES: pytest numpy
    CIBW_TEST_COMMAND: pytest {project}/tests
  run: |
    pip install --upgrade cibuildwheel
    python -m cibuildwheel --output-dir wheelhouse
```

and for Mac, it's just:

```yaml
- name: Build wheels
  env:
    CIBW_SKIP: pp*
    CIBW_TEST_REQUIRES: pytest numpy
    CIBW_TEST_COMMAND: pytest {project}/tests
  run: |
    pip install --upgrade cibuildwheel
    python -m cibuildwheel --output-dir wheelhouse
```

**Comment by henryiii** [comment]
Is this all versions of Windows, or just 3.5 (which will be the first one it tries)? I've noticed issues getting setuptools_scm to work with 3.5 on Windows, but I ended up just dropping 3.5 on Windows (and macOS) since those are probably quite rare. This might have indicated a pip version or setuptools version issue, though.

**Comment by Czaki** [comment]
In my project `pyproject.toml` works. But I do not have python 3.5 build. So I also think that this is problem with setuptools/pip.

**Comment by henryiii** [comment]
@ajfriend Could you try changing this line:

```diff
-    CIBW_SKIP: pp* cp27* *-win32
+    CIBW_SKIP: pp* cp27* *-win32 cp35-win_amd64
-    CIBW_BEFORE_BUILD: pip install scikit-build cython cmake
```

And dropping the `CIBW_BEFORE_BUILD` in a PR? Does that pass?

**Comment by ajfriend** [comment]
That worked: https://github.com/uber/h3-py/actions/runs/114979866

What is it about Python 3.5 that's causing this?

**Comment by henryiii** [comment]
I wasn't able to experiment properly with boost-histogram, because the default compiler on Windows for 3.5 isn't quite up to compiling C++14 (IIRC), so I didn't have an old 3.5 build. In fact, I was surprised it worked at all, except for the setuptools_scm not picking up the version number correctly. I think it will be easier to debug in h3-py.

**Comment by henryiii** [comment]
I do notice there's an old pip 9 laying around in that env, but it supposedly is being replaced / superseded by the pinned version, which is 20.something, which should respect the PEP 517 build.

**Comment by henryiii** [comment]
The tests in the h3 repo show that pip is correctly installing the PEP 517 requirements, then not using the environment when it runs the build.

@joerick How hard would it be to add a PEP 517 build test to the test folder? I'm going to predict it will fail on Python 3.5 + Windows. Possibly a bug in Pip, though, or maybe a very unlucky issue with Python 3.5.4 (I see a path handling change in 3.5.5, don't think it would cause this, but I am not sure).

The `pyproject.toml` should add one non-standard package (requests or something, doesn't matter), then the setup.py should simply verify that that package is installed when it runs.

* pyproject.toml
```toml
[build-system]
requires = [
    "setuptools>=42",
    "wheel"
]

build-backend = "setuptools.build_meta"
```

* setup.py
```python
import requests

# + minimum required for package to run wheel build process
```

**Comment by joerick** [comment]
I think such a test would be useful. If you have time, please send a PR!

**Comment by joerick** [comment]
Fixed in #358.
